src/app/core/layers/map/vectorlayer.js

```diff
@@ -86,14 +86,14 @@ proto._makeStyle = function(styleConfig) {
 };
 
 proto.getFeatures = function(options={}) {
-  const d = $.Deferred();
-  this.provider.getFeatures(options)
-    .then(features => {
-      this.addFeatures(features);
-      d.resolve(features);
-    })
-    .fail(err => d.reject(err));
-  return d.promise()
+  return new Promise((resolve, reject) =>{
+    this.provider.getFeatures(options)
+      .then(features => {
+        this.addFeatures(features);
+        resolve(features);
+      })
+      .catch(err => reject(err));
```

Comment:
Not sure what the previous behavior was, but now I think you need to add the return `(resolve, reject) => this.provider.getFeatures` from inside the promise to the outside to make `. catch(err => reject(err))` is trappable from outside (otherwise it will only throw an unhandled fatal error):

```suggestion
return new Promise((resolve, reject) =>
    this.provider
      .getFeatures(options)
      .then(features => resolve(this.addFeatures(features)))
      .catch(err => reject(err))
);
```

Anyway i'm not so sure about this statement, check it carefully too